### ISSUE TITLE
Cannot reproduce the training curves of LLaMA3-simpo-v2

### ISSUE THREAD

**Issue by jdf-prog** [princeton-nlp/SimPO#60]
I tried to reproduce the simpo-v2's results on llama-3, but find the training curve is quite different from your released version. I am using 8x A6000 GPUs.
Here is the training config:
```
# Model arguments
model_name_or_path: meta-llama/Meta-Llama-3-8B-Instruct
torch_dtype: null
attn_implementation: flash_attention_2

# Data training arguments
dataset_mixer:
  princeton-nlp/llama3-ultrafeedback-armorm: 1.0
dataset_splits:
- train
- test
preprocessing_num_workers: 12

# SimPOTrainer arguments
bf16: true
beta: 10
gamma_beta_ratio: 0.3
do_eval: true
evaluation_strategy: steps
eval_steps: 400
gradient_accumulation_steps: 8
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: False
hub_model_id: simpo-exps
learning_rate: 1.0e-6
log_level: info
logging_steps: 1
lr_scheduler_type: cosine
max_length: 2048
max_prompt_length: 1800
num_train_epochs: 1
optim: adamw_torch
output_dir: outputs/llama-3-8b-instruct-simpo-v2
run_name: llama-3-8b-instruct-simpo-v2
per_device_train_batch_size: 2
per_device_eval_batch_size: 4
push_to_hub: false
save_strategy: "steps"
save_steps: 1000000
report_to:
- wandb
save_total_limit: 20
seed: 42
warmup_ratio: 0.1

```
[Here is the my training curve](https://wandb.ai/dongfu/huggingface/runs/4dheywn9?nw=nwuserdongfu)
![image](https://github.com/user-attachments/assets/5f4598ac-e3c7-4ca6-897c-3b4ae1b49210)

[And your released training curve](https://wandb.ai/yumeng0818/simpo/runs/zvv56fcj?nw=nwuseryumeng0818)
<img width="1693" alt="image" src="https://github.com/user-attachments/assets/74f95e51-ffd8-40bd-a652-a85f67759bea">

I am wondering how to fix this issue.

**Comment by jdf-prog** [comment]
It turns out to be the problem of wandb x axis. I need to change it to global steps instead of logging steps in the simpo released wandb curve.
![image](https://github.com/user-attachments/assets/a2ab2873-d878-45e2-a33f-06076c3f2fda)
Then the loss curve is reasonable, looks like this:

![image](https://github.com/user-attachments/assets/223f8b40-8d09-4a72-a1b8-204b00e8ac80)

**Comment by jdf-prog** [comment]
I also tried using llamafactory instead of the simpo offical codes for the training, it turns out I can get similar training curves for full training. But when changing to lora using llamafactory, with the exact same hyper parameters as full training, the loss curve looks weird. Probably because of lora and full fine-tuning of SimPO/DPO need different hyper parameters. For example, lora might require a higher learning rate. 

My curves:
[https://wandb.ai/dongfu/huggingface/reports/Reproduction-of-SimPO-V2-of-LLaMA-3-8B-Instruct--Vmlldzo4OTU5Mjk3](https://wandb.ai/dongfu/huggingface/reports/Reproduction-of-SimPO-V2-of-LLaMA-3-8B-Instruct--Vmlldzo4OTU5Mjk3)
